forecasttools/data.py

```diff
@@ -114,365 +99,93 @@ def make_nshn_fitting_dataset(
-        # get the state data
-        state_nhsn = nhsn.filter(pl.col("state") == state)
-        # get actual data y, X
-        y = state_nhsn["hosp"].to_numpy()[0]
-        X = np.arange(y.shape[0])
-        # set up inference, NUTS/MCMC
-        kernel = numpyro.infer.NUTS(model=model)
-        mcmc = numpyro.infer.MCMC(
-            kernel, num_warmup=num_warmup, num_samples=num_samples
-        )
-        # create spline basis for obs period and forecast period
-        last = X[-1]
-        X_future = np.hstack(
-            (X, np.arange(last + 1, last + 1 + forecast_days))
-        )
-        sbm = spline_basis(X_future)
-        # get posterior samples
-        mcmc.run(rng_key=jr.key(random_seed), basis_matrix=sbm[: len(X)], y=y)
-        posterior_samples = mcmc.get_samples()
-        # get prior predictive
-        prior_pred = numpyro.infer.Predictive(model, num_samples=num_samples)(
-            rng_key=jr.key(random_seed), basis_matrix=sbm[: len(X)]
-        )
-        # get posterior predictive forecast
-        posterior_pred_for = numpyro.infer.Predictive(
-            model, posterior_samples=posterior_samples
-        )(rng_key=jr.key(random_seed), basis_matrix=sbm)
-        # create initial inference data object(s)
-        idata = az.from_numpyro(
-            posterior=mcmc,
-            posterior_predictive=posterior_pred_for,
-            prior=prior_pred,
-        )
-        # save idata object(s)
-        idata.to_netcdf(file_save_path)
-        # plot forecast (if desired) from idata light
-        if show_plot:
-            x_data = idata.posterior_predictive["obs_dim_0"]
-            y_data = idata.posterior_predictive["obs"]
-            fig, axes = plt.subplots(1, 1, figsize=(8, 6))
-            az.plot_hdi(
-                x_data,
-                y_data,
-                hdi_prob=0.95,
-                color="skyblue",
-                smooth=False,
-                fill_kwargs={"alpha": 0.2, "label": "95% Credible"},
-                ax=axes,
-            )
-            az.plot_hdi(
-                x_data,
-                y_data,
-                hdi_prob=0.75,
-                color="skyblue",
-                smooth=False,
-                fill_kwargs={"alpha": 0.4, "label": "75% Credible"},
-                ax=axes,
-            )
-            az.plot_hdi(
-                x_data,
-                y_data,
-                hdi_prob=0.5,
-                color="C0",
-                smooth=False,
-                fill_kwargs={"alpha": 0.6, "label": "50% Credible"},
-                ax=axes,
+    else:
+        # check that the loaded CSV has the needed columns
+        df_cols = pl.scan_csv(nhsn_dataset_path).columns
+        required_cols = [
+            "state",
+            "date",
+            "previous_day_admission_adult_covid_confirmed",
+            "previous_day_admission_influenza_confirmed",
+        ]
+        if len(set(required_cols).intersection(set(df_cols))) != len(
```

Comment:
What is this doing? based on the comment, it seems like you just want to check if the required columns are a subset of the given colums. If so, why not use [`issubset`](https://docs.python.org/3/library/stdtypes.html#frozenset.issubset)?

Reply:
Good catch. If you look at other checks of this sort in the code `issubset` is used. I think I missed changing this once I saw there was, in fact, an `issubset` method.